### ISSUE TITLE
`vsphere-iso`: convert to template in content library error - 404 Not Found

### ISSUE THREAD

**Issue by msilveirabr** [hashicorp/packer-plugin-vsphere#238]
Packer library content convert works perfectly fine with vCenter 6.7.

In vCenter 6.5 I'm getting these errors:
```
2022/11/17 17:09:04 ui: 2022-11-17T17:09:04-03:00: ==> vsphere-iso.linux-fedora-server-fc37: VM is already powered off
2022/11/17 17:09:04 ui: 2022-11-17T17:09:04-03:00: ==> vsphere-iso.linux-fedora-server-fc37: Deleting Floppy drives...
2022/11/17 17:09:04 ui: 2022-11-17T17:09:04-03:00: ==> vsphere-iso.linux-fedora-server-fc37: Eject CD-ROM drives...
2022/11/17 17:09:04 ui: 2022-11-17T17:09:04-03:00: ==> vsphere-iso.linux-fedora-server-fc37: Deleting CD-ROM drives...
2022/11/17 17:09:05 ui: 2022-11-17T17:09:05-03:00: ==> vsphere-iso.linux-fedora-server-fc37: Clear boot order...
2022/11/17 17:09:05 ui: 2022-11-17T17:09:05-03:00: ==> vsphere-iso.linux-fedora-server-fc37: Importing VM template linux-fedora-server-fc37-en_US-v20221117.16591668715166 to Content Library...
2022/11/17 17:09:06 ui error: 2022-11-17T17:09:06-03:00: ==> vsphere-iso.linux-fedora-server-fc37: Failed to import template linux-fedora-server-fc37-en_US-v20221117.16591668715166: POST https://vcenter01.localdomain/rest/vcenter/vm-template/library-items: 404 Not Found
2022/11/17 17:09:06 ui error: 2022-11-17T17:09:06-03:00: ==> vsphere-iso.linux-fedora-server-fc37: POST https://vcenter01.localdomain/rest/vcenter/vm-template/library-items: 404 Not Found
2022/11/17 17:09:06 ui: 2022-11-17T17:09:06-03:00: ==> vsphere-iso.linux-fedora-server-fc37: Step "StepImportToContentLibrary" failed
```
According to packer builder documentation, vCenter 6.5 should be supported:
https://developer.hashicorp.com/packer/plugins/builders/vsphere/vsphere-iso

My build is based on vsphere-iso example: https://github.com/vmware-samples/packer-examples-for-vsphere

my config/vcenter.pkrvars.hcl:
```hcl
// Template and Content Library Settings for vCenter API
common_template_conversion         = false
common_content_library_ovf         = false
common_content_library_destroy     = true
common_content_library_skip_import = false
common_content_library_name        = "VMs"
```
My linux-fedora.pkr.hcl contains:

```hcl
  // Template and Content Library Settings
  convert_to_template = var.common_template_conversion
  dynamic "content_library_destination" {
    for_each = var.common_content_library_name != null ? [1] : []
    content {
      library     = var.common_content_library_name
      description = "Version: v${local.build_version}\nBuilt on: ${local.build_date}\n${local.build_by}"
      ovf         = var.common_content_library_ovf
      destroy     = var.common_content_library_destroy
      skip_import = var.common_content_library_skip_import
    }
  }
```

As stated in the first paragraph: It is working fine with vCenter 6.7 .

**Comment by tenthirtyam** [comment]
General support for vSphere 6.5 and 6.7 ended on October 15, 2022. All features starting with v1.1.0 are only tested on vSphere 7.0 and later.

If you need help with a version prior to vSphere 7.0, we ask that you reach out to the community via the [Packer discussion forums](https://discuss.hashicorp.com/c/packer/23). 

Any support questions asked on the GitHub tracker will be handled on a best effort basis. 

If there is no further activity on this question within the next 30 days it will be closed.

>**Note**
>
> Please note that the README.md for the plugins has been updated, but there is still a need to update the online documentation.

**Ryan Johnson**
Senior Staff Solutions Architect | Product Engineering @ VMware, Inc.

**Comment by tenthirtyam** [comment]
I’m also the maintainer of the project you’ve referenced, which is **only** tested for vSphere 7.0 U3D and later due to an issue for the content library that was resolved in 7.0 U2.

if using the tested versions, the correct syntax in the common variables would be:

```hcl
// Template and Content Library Settings
common_template_conversion         = false
common_content_library_name        = "sfo-w01-lib01"
common_content_library_ovf         = true
common_content_library_destroy     = true
common_content_library_skip_export = false
```

**Ryan Johnson**
Senior Staff Solutions Architect | Product Engineering @ VMware, Inc.

**Comment by tenthirtyam** [comment]
Lastly, it should be mentioned that OVF templates can be imported to the content library in vSphere 6.5 and later; however, support for VM Templates was added in vSphere 6.7 U1.

Reference: https://docs.vmware.com/en/VMware-vSphere/6.7/rn/vsphere-vcenter-server-671-release-notes.html
